test/python/quantum_info/operators/test_operator.py

```diff
@@ -1050,6 +1051,113 @@ def test_from_circuit_mixed_reg_loose_bits_transpiled(self):
+        op = Operator(self.rand_matrix(64, 64))
+        pattern = [1, 2, 0, 3, 5, 4]
+
+        # Consider several methods of computing this operator and show
+        # they all lead to the same result.
+
+        # Compose the operator with the operator constructed from the
+        # permutation circuit.
+        op2 = op.copy()
+        perm_op = Operator(Permutation(6, pattern))
+        op2 &= perm_op
+
+        # Compose the operator with the operator constructed from the
+        # permutation gate.
+        op3 = op.copy()
+        perm_op = Operator(PermutationGate(pattern))
+        op3 &= perm_op
+
+        # Modify the operator using apply_permutation method.
+        op4 = op.copy()
+        op4 = op4.apply_permutation(pattern, front=False)
+
+        self.assertEqual(op2, op3)
+        self.assertEqual(op2, op4)
+
+    def test_apply_permutation_front(self):
+        """Test applying permutation to the operator,
+        where the permutation is applied first and the operator second"""
+        op = Operator(self.rand_matrix(64, 64))
+        pattern = [1, 2, 0, 3, 5, 4]
+
+        # Consider several methods of computing this operator and show
+        # they all lead to the same result.
+
+        # Compose the operator with the operator constructed from the
+        # permutation circuit.
+        op2 = op.copy()
+        perm_op = Operator(Permutation(6, pattern))
+        op2 = perm_op & op2
+
+        # Compose the operator with the operator constructed from the
+        # permutation gate.
+        op3 = op.copy()
+        perm_op = Operator(PermutationGate(pattern))
+        op3 = perm_op & op3
+
+        # Modify the operator using apply_permutation method.
+        op4 = op.copy()
+        op4 = op4.apply_permutation(pattern, front=True)
+
+        self.assertEqual(op2, op3)
+        self.assertEqual(op2, op4)
+
+    def test_reverse_qargs_as_apply_permutation(self):
+        """Test reversing qargs by pre- and post-composing with reversal
+        permutation.
+        """
+        perm = [3, 2, 1, 0]
+
+        for dims in [
+            (2, 3, 4, 5),
+            (5, 2, 4, 3),
+            (3, 5, 2, 4),
+            (5, 3, 4, 2),
+            (4, 5, 2, 3),
+            (4, 3, 2, 5),
+        ]:
+            op = Operator(
+                np.array(range(120 * 120)).reshape((120, 120)), input_dims=dims, output_dims=dims
+            )
+            op2 = op.reverse_qargs()
+            op3 = op.apply_permutation(perm, front=True).apply_permutation(perm, front=False)
+            self.assertEqual(op2, op3)
+
+    def test_apply_permutation_exceptions(self):
+        """Checks that applying permutation raises an error when dimensions do not match."""
+        op = Operator(
+            np.array(range(24 * 30)).reshape((24, 30)), input_dims=(6, 5), output_dims=(2, 3, 4)
+        )
+
+        # These are ok
+        op.apply_permutation([2, 1, 0], front=False)
+        op.apply_permutation([1, 0], front=True)
```

Comment:
We probably don't need to test these here, since the test is about the exceptions.  We could test their full validity in a separate test, though.

Reply:
Done.